### PR TITLE
Added config option for base dir

### PR DESCRIPTION
**Pull request by Bavoen**

### CODE REVIEW

**Comment by acopet** [comment]
Hello @Bavoen,

First of all, thank you for this contribution.

In order to integrate this pull-request, could you please explain the need behind this change? Is it linked to the issue reported through this ticket #25?

Cheers,
Antoine

**Comment by Bavoen** [comment]
Hello,

Well, i'm not really sure whether it will fix that specific problem since we experienced an other difficulty, but it might.
The problem we had was that our maven pom file was not located in the root of our repository but in a separate project. This plugin takes the place of the pom as it's home directory and later tries to generate the relative path for some files using that home directory. Since our project looked something like this:

A
|- Build
\-B

The home directory is set to A/Build and the code tries to match A/Build with A/B coming up with nothing. Therefor we added the option to make it possible to add a home dir (in this case we would set it to A).
I hope this makes our case clear.

I have noticed some funny things happend to this pull request, I will try to update it as soon as possible. 
I just noticed that `config.getBaseDir() == ""` might not be a good test either

**Comment by acopet** [comment]
Thanks for this clarification, we will review this change on this week.

Cheers,
Antoine

**Comment by nvillard** [comment]
Hi @Bavoen ,

This is indeed a feature we want to implement as well for our usage.
What I can say having a quick look at the diff is no test has been included.
It looks like coding guidelines are not followed.
Feel free to update the pull request with those updates. However, I will take it and update it accordingly in the coming weeks.
Thanks a lot again for your contribution.
Nicolas

**Comment by chutch** [comment]
Hi @Bavoen, @nvillard 

is there any progress on that issue? I would also take advantage of that feature and I'm wondering if it's going to get into official release anytime soon.

Great work BTW!

File: src/main/java/org/sonar/plugins/stash/StashRequestFacade.java [hunk 1]
```diff
@@ -40,7 +40,14 @@ public StashRequestFacade(StashPluginConfiguration stashPluginConfiguration) {
   }
   
   public void initialize(File projectBaseDir) {
-    this.projectBaseDir = projectBaseDir;
+    if (config.getBaseDir() != null && !"".equals(config.getBaseDir()))
```
Comment (t-8ch) [inline src/main/java/org/sonar/plugins/stash/StashRequestFacade.java]: * Is this the correct logic? We should use the basedir from the configuration if it exists, or?
* `StringUtils.isNotEmpty()`/`StringUtils.isEmpty()`
* Please keep the code style consistent (no extra lines for braces)
* This method could be renamed to closer match its function: `setDefaultBaseDir` (open to suggestions)

**Comment by t-8ch** [comment]
@nvillard @Bavoen @chutch 
Could you test the plugin with this patch and give feedback whether it works for you?
(You should probably invert the logic in `StashRequestFacade.java:43`)
This makes it so much easier for me to merge this.
Thanks!

**Comment by kirkor** [comment]
Hi,
I've test this change on my testing project: https://github.com/kirkor/sonar-stash-test-project
It's working.

In addition I've push requested changes from this PR to my repo: https://github.com/kirkor/sonar-stash

This PR looks dead, so I'm creating new one with this change + mine:

**Comment by kirkor** [comment]
This problem can be solved by proper configuration. I've describe it in PR #56 and here: https://github.com/kirkor/sonar-stash-test-project/blob/master/README.md

**Comment by t-8ch** [comment]
See #56

**Pull request closed without merging**